code/modules/mob/living/carbon/human/examine.dm

```diff
@@ -380,6 +380,39 @@
 
 	to_chat(user, msg.Join())
 
+	if(istype(user, /mob/living/carbon/human))
+		var/mob/living/carbon/human/U = user
+		if(U.species.can_make_eye_contact && distance < 4)
+			var/eyescovered = FALSE
+			for(var/obj/item/clothing/C in list(U.wear_mask, U.head, U.glasses))
+				if(C.body_parts_covered & EYES)
+					eyescovered = TRUE
+					break
+			if(!eyescovered)
+				U.handle_examine(src)
+				U.recent_examines += src //Done after so that it doesn't work with itself
```

Comment:
I don't think we should hold hardrefs to mobs? Maybe a weakref here would be good instead

Reply:
Still working on this one with help from Geeves. It seems to not quite agree on being saved properly as a weakref.